<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>5.SNV correction â€¢ MAAS</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="5.SNV correction">
<meta property="og:description" content="MAAS">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAAS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-atom"></span>
     
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/1.scATAC-seq_data_analysis.html">1.scATAC-seq data analysis</a>
    </li>
    <li>
      <a href="../articles/2.CNV%20calling.html">2.CNV calling</a>
    </li>
    <li>
      <a href="../articles/3.SNV%20calling.html">3.SNV calling</a>
    </li>
    <li>
      <a href="../articles/4.Peak%20correction.html">4.Peak correction</a>
    </li>
    <li>
      <a href="../articles/5.SNV_correction.html">5.SNV correction</a>
    </li>
    <li>
      <a href="../articles/6.MAAS.html">6.MAAS</a>
    </li>
    <li>
      <a href="../articles/7.Clustering.html">7.Clustering</a>
    </li>
    <li>
      <a href="../articles/8.Functional_enrichment.html">8.Functional enrichment</a>
    </li>
  </ul>
</li>
<li>
  <a href="../articles/Contact.html">Contact</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>5.SNV correction</h1>
            
      
      
      <div class="hidden name"><code>5.SNV_correction.Rmd</code></div>

    </div>

    
    
<p>We have conducted benchmark of several SNV denoising tools designed
for single-cell data, and found that CBM show superior performance
(please refer to our paper for more details). Users should install and
run <a href="https://github.com/zhyu-lab/cbm" class="external-link">CBM</a> before running
MAAS.<br></p>
<p>In this tutorial, CBM has been installed in
<code>~/App/cbm-main</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">input_file</span> <span class="op">&lt;-</span> <span class="st">"snv.CBMinput.txt"</span></span></code></pre></div>
<p>The raw SNV profile as input for CBM is a cell-by-mutation matrix
(cells in rows and mutations in columns). The cell and mutation names
will be recovered after executing CBM.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"http://bioinfo.szbl.ac.cn/share/MAAS_data/snv.CBMinput.txt"</span>, sep <span class="op">=</span> <span class="st">'\t'</span>, header <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">input</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;    X1 X0 X0.1 X0.2 X0.3 X0.4 X0.5 X0.6 X0.7 X0.8</span></span>
<span><span class="co">#&gt; 1   0  0    1    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; 2   0  0    0    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; 3   0  0    0    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; 4   0  0    0    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; 5   0  0    0    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; 6   0  0    0    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; 7   0  0    0    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; 8   0  0    0    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; 9   0  0    0    0    0    0    0    0    0    0</span></span>
<span><span class="co">#&gt; 10  0  0    0    0    0    0    0    0    0    0</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># sample.tmp &lt;- gsub("\\.snv\\.CBMinput\\.txt", "", file.tmp)</span></span>
<span><span class="va">output_file</span> <span class="op">&lt;-</span> <span class="st">"./"</span></span>
<span><span class="va">cmd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"~/App/cbm-main/bin/cbm -i"</span>, <span class="va">input_file</span>, <span class="st">"-o"</span>, <span class="va">output_file</span>, <span class="st">"-K 10"</span><span class="op">)</span> <span class="co"># Set up to 10 clones</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.html" class="external-link">system</a></span><span class="op">(</span><span class="va">cmd</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pred.cluster.genotype</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">".clones.txt"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pred.cluster.genotype</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">0</span><span class="op">:</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pred.cluster.genotype</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">pred.cluster.genotype</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span></code></pre></div>
<p>The genotype of each cluster. Clusters in rows and mutations in
columns</p>
<pre><code><span><span class="co">#&gt;   V1 V2 V3 V4 V5 V6 V7 V8 V9 V10</span></span>
<span><span class="co">#&gt; 0  1  0  1  1  0  0  0  0  0   0</span></span>
<span><span class="co">#&gt; 1  0  0  0  0  0  0  0  0  0   0</span></span>
<span><span class="co">#&gt; 2  0  0  0  0  0  0  0  0  0   0</span></span>
<span><span class="co">#&gt; 3  0  0  0  0  0  0  0  0  0   0</span></span>
<span><span class="co">#&gt; 4  0  0  0  0  0  0  0  0  0   0</span></span>
<span><span class="co">#&gt; 5  0  0  0  0  0  0  0  0  0   0</span></span>
<span><span class="co">#&gt; 6  0  0  0  0  0  0  0  0  0   0</span></span>
<span><span class="co">#&gt; 7  0  0  0  0  0  0  0  0  1   1</span></span>
<span><span class="co">#&gt; 8  0  0  0  0  0  0  0  0  0   0</span></span>
<span><span class="co">#&gt; 9  1  0  1  1  0  0  0  0  0   1</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pred.cluster.label</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">".assignment.txt"</span><span class="op">)</span></span>
<span><span class="va">pred.cluster.label</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">pred.cluster.label</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pred.cluster.label</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"Cluster"</span> <span class="co"># One-dimension vector: Cells in rows with assigned clusters</span></span>
<span><span class="va">pred.cluster.label</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,,drop <span class="op">=</span> <span class="cn">F</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;     Cluster</span></span>
<span><span class="co">#&gt; V1        0</span></span>
<span><span class="co">#&gt; V2        1</span></span>
<span><span class="co">#&gt; V3        2</span></span>
<span><span class="co">#&gt; V4        3</span></span>
<span><span class="co">#&gt; V5        4</span></span>
<span><span class="co">#&gt; V6        4</span></span>
<span><span class="co">#&gt; V7        3</span></span>
<span><span class="co">#&gt; V8        3</span></span>
<span><span class="co">#&gt; V9        4</span></span>
<span><span class="co">#&gt; V10       3</span></span></code></pre>
<p>The cells from the same clone are assumed to be same. Hence, we
assign cluster-level mutations to individual cells</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pred</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">pred.cluster.label</span><span class="op">$</span><span class="va">Cluster</span>, <span class="kw">function</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">pred.cluster.genotype</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pred</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;                  chr1:634045:T&gt;G chr16:46390534:G&gt;A chr17:22521407:G&gt;A</span></span>
<span><span class="co">#&gt; AAACTCGCATTCTTTG               1                  0                  1</span></span>
<span><span class="co">#&gt; AAACTCGTCCATTGTT               0                  0                  0</span></span>
<span><span class="co">#&gt; AAACTCGTCGTTGTTT               0                  0                  0</span></span>
<span><span class="co">#&gt; AAACTGCAGAGTCCGA               0                  0                  0</span></span>
<span><span class="co">#&gt; AAACTGCCAGGGAGTT               0                  0                  0</span></span>
<span><span class="co">#&gt; AAACTGCTCTACATCT               0                  0                  0</span></span>
<span><span class="co">#&gt; AAAGATGCAGGTGTCC               0                  0                  0</span></span>
<span><span class="co">#&gt; AAAGATGGTCCGTGCA               0                  0                  0</span></span>
<span><span class="co">#&gt; AAAGGATGTTCCGCGA               0                  0                  0</span></span>
<span><span class="co">#&gt; AAAGGGCAGACCAATA               0                  0                  0</span></span>
<span><span class="co">#&gt;                  chr17:22521415:T&gt;C chr17:22521426:T&gt;C chr22:10742083:C&gt;T</span></span>
<span><span class="co">#&gt; AAACTCGCATTCTTTG                  1                  0                  0</span></span>
<span><span class="co">#&gt; AAACTCGTCCATTGTT                  0                  0                  0</span></span>
<span><span class="co">#&gt; AAACTCGTCGTTGTTT                  0                  0                  0</span></span>
<span><span class="co">#&gt; AAACTGCAGAGTCCGA                  0                  0                  0</span></span>
<span><span class="co">#&gt; AAACTGCCAGGGAGTT                  0                  0                  0</span></span>
<span><span class="co">#&gt; AAACTGCTCTACATCT                  0                  0                  0</span></span>
<span><span class="co">#&gt; AAAGATGCAGGTGTCC                  0                  0                  0</span></span>
<span><span class="co">#&gt; AAAGATGGTCCGTGCA                  0                  0                  0</span></span>
<span><span class="co">#&gt; AAAGGATGTTCCGCGA                  0                  0                  0</span></span>
<span><span class="co">#&gt; AAAGGGCAGACCAATA                  0                  0                  0</span></span>
<span><span class="co">#&gt;                  chr22:10742088:G&gt;T chr3:93470405:A&gt;T chr3:93470446:G&gt;A</span></span>
<span><span class="co">#&gt; AAACTCGCATTCTTTG                  0                 0                 0</span></span>
<span><span class="co">#&gt; AAACTCGTCCATTGTT                  0                 0                 0</span></span>
<span><span class="co">#&gt; AAACTCGTCGTTGTTT                  0                 0                 0</span></span>
<span><span class="co">#&gt; AAACTGCAGAGTCCGA                  0                 0                 0</span></span>
<span><span class="co">#&gt; AAACTGCCAGGGAGTT                  0                 0                 0</span></span>
<span><span class="co">#&gt; AAACTGCTCTACATCT                  0                 0                 0</span></span>
<span><span class="co">#&gt; AAAGATGCAGGTGTCC                  0                 0                 0</span></span>
<span><span class="co">#&gt; AAAGATGGTCCGTGCA                  0                 0                 0</span></span>
<span><span class="co">#&gt; AAAGGATGTTCCGCGA                  0                 0                 0</span></span>
<span><span class="co">#&gt; AAAGGGCAGACCAATA                  0                 0                 0</span></span>
<span><span class="co">#&gt;                  chr3:93470451:C&gt;T</span></span>
<span><span class="co">#&gt; AAACTCGCATTCTTTG                 0</span></span>
<span><span class="co">#&gt; AAACTCGTCCATTGTT                 0</span></span>
<span><span class="co">#&gt; AAACTCGTCGTTGTTT                 0</span></span>
<span><span class="co">#&gt; AAACTGCAGAGTCCGA                 0</span></span>
<span><span class="co">#&gt; AAACTGCCAGGGAGTT                 0</span></span>
<span><span class="co">#&gt; AAACTGCTCTACATCT                 0</span></span>
<span><span class="co">#&gt; AAAGATGCAGGTGTCC                 0</span></span>
<span><span class="co">#&gt; AAAGATGGTCCGTGCA                 0</span></span>
<span><span class="co">#&gt; AAAGGATGTTCCGCGA                 0</span></span>
<span><span class="co">#&gt; AAAGGGCAGACCAATA                 0</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw.snv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"snv.mat.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pred</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">raw.snv</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pred</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">raw.snv</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">pred</span>, <span class="st">"denoised.mat.rds"</span><span class="op">)</span></span></code></pre></div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Kewei Xiong.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
